rr_viz/src/ross/mission_editor/mission_editor.py

```diff
@@ -0,0 +1,201 @@
+
+        # Command buttons
+        self.h_layout_command = QHBoxLayout()
+
+        self.go_to_selected_button = QPushButton('Go to Selected')
+        self.go_to_selected_button.pressed.connect(self.go_to_selected)
+        
+        self.go_to_all_button = QPushButton('Go to All')
+        self.go_to_all_button.pressed.connect(self.go_to_all)
+
+        self.enable_go_to_buttons(False)
+        self.set_enable_go_to_buttons.connect(self.enable_go_to_buttons)
+
+        self.send_mission_button = QPushButton('Send Mission')
+        self.send_mission_button.pressed.connect(self.send_mission)
+        self.enable_send_mission_button(False)
+        self.set_enable_send_mission.connect(self.enable_send_mission_button)
+
+        self.h_layout_command.addWidget(self.go_to_selected_button)
+        self.h_layout_command.addWidget(self.go_to_all_button)
+        self.h_layout_command.addWidget(self.send_mission_button)
+
+        self.v_layout.addLayout(self.h_layout_command)
+
+        self.setLayout(self.v_layout)
+
+        self.bt_action = BuildBTAction()
+        self.mb_action = WaypointMoveBaseAction()
+        
+        # Setup state checker:
+        self.bt_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.bt_action.is_connected,  self.set_enable_send_mission, Hz=1./3.)
+        self.bt_state_checker.start()
+        self.mb_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.mb_action.is_connected,  self.set_enable_go_to_buttons, Hz=1./3.)
+        self.mb_state_checker.start()
+
+    def robot_pose_cb(self, msg):
+        self.robot_pose = msg
+
+    def spawn_waypoint(self, pwcs_msg):
+        msg = TaskWaypointMsg()
+        pose_stamped = PoseStamped()
+        pose_stamped.header = pwcs_msg.header
+        pose_stamped.pose = pwcs_msg.pose.pose
+        msg.pose_stamped = pose_stamped
+        self.waypoint_list.append(self.waypoint_list.new_waypoint(msg))
+
+    def add_here(self):
+        if self.robot_pose:
+            msg = PoseWithCovarianceStamped()
+            msg.header = self.robot_pose.header
+            msg.pose.pose = self.robot_pose.pose
+            self.spawn_waypoint_signal.emit(msg)
+        else:
+            rospy.logwarn("Robot pose not connected")
+
+    def duplicate(self):
+        if self.waypoint_list.get_selected_wp():
+            self.waypoint_list.duplicate(self.waypoint_list.get_selected_wp())
+
+    def delete(self):
+        if self.waypoint_list.get_selected_wp():
+            self.waypoint_list.remove(self.waypoint_list.get_selected_wp())        
+
+    def delete_all(self):
+        for _ in range(self.waypoint_list.len()):
+            self.waypoint_list.remove(
+                self.waypoint_list.get_wp(self.waypoint_list.len()-1))
+
+    def save(self):
+        mission_name, ok = QInputDialog.getText(self, "Mission file name", "Specify file name to save mission:")
+
+        if ok:
+            if mission_name[-5:] == ".yaml":
```

Comment:
what happens if the user provides text "ab" here? 

I would probably use something like `if mission_name.enswith(".yaml")...`

Low effort approach (not tested):
```
if mission_name.endswith(".yaml"):
   mission_name = mission_name.replace(".yaml", "")
```

Reply:
I just checked and if you put 'ab' there it adds on '.yaml' to the end. And if you put 'ab.yaml' it doesn't modify it